File: python/39 LoopPatterns.py

def printRightHalfPyramid(rows):
    for i in range(rows):
        for j in range(i+1):
            print("*",end="")
        print("")
def printReverseRightHalfPyramid(rows):
    for i in range(rows):
        for j in range(rows-i,0,-1):
            print("*",end="")
        print("")
def printLeftHalfPyramid(rows):
    for i in range(rows):
        for j in range(rows-i,1,-1):
            print(" ",end="")
        for k in range(i+1):
            print("*",end="")
        print("")

rows = int(input("Enter number of rows: "))
print("Right Half Pyramid:")
printRightHalfPyramid(rows)
print("Reverse Right Half Pyramid:")
printReverseRightHalfPyramid(rows)
print("Left Half Pyramid:")
printLeftHalfPyramid(rows)

File: java/39 LoopPatterns.java
package code;
import java.util.Scanner;
public class Test {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter nummber of rows: ");
		int rows=sc.nextInt();
		System.out.println("Right Half Pyramid:");
		printRightHalfPyramid(rows);
		System.out.println("Reverse Right Half Pyramid:");
        printReverseRightHalfPyramid(rows);
        System.out.println("Left Half Pyramid:");
        printLeftHalfPyramid(rows);

	}
	 static void printLeftHalfPyramid(int rows) {
		 for(int i=0;i<rows;i++) {
				for(int j=rows-1-i;j>0;j--) {
					System.out.print(" ");
				}
				for(int k=0;k<i+1;k++) {
					System.out.print("*");
				}
				System.out.println("");
			}
	}
	 static void printReverseRightHalfPyramid(int rows) {
		for(int i=0;i<rows;i++) {
			for(int j=rows-i;j>0;j--) {
				System.out.print("*");
			}
			System.out.println("");
		}
		
	}
	 static void printRightHalfPyramid(int rows) {
		for(int i=0;i<=rows;i++) {
			for(int j=0;j<i;j++) {
				System.out.print("*");
			}
			System.out.println("");
		}
	}
}
